packages/core/lib/metadata-delegate.js

```diff
@@ -0,0 +1,53 @@
+module.exports = {
+  add (section, ...args) {
+    if (!section) return this
+    let updates
+
+    // addMetadata("section", null) -> clears section
+    if (args[0] === null) return this.clearMetadata(section)
+
+    // normalise the two supported input types into object form
+    if (typeof args[0] === 'object') updates = args[0]
+    if (typeof args[0] === 'string') updates = { [args[0]]: args[1] }
+
+    // exit if we don't have an updates object at this point
+    if (!updates) return this
+
+    // ensure a section with this name exists
+    if (!this._metadata[section]) this._metadata[section] = {}
+
+    // merge the updates with the existing section
+    this._metadata[section] = { ...this._metadata[section], ...updates }
+
+    return this
+  },
+
+  get (section, key) {
+    if (typeof section !== 'string') return undefined
+    if (!key) {
+      return this._metadata[section]
+    }
+    if (this._metadata[section]) {
+      return this._metadata[section][key]
+    }
+    return undefined
+  },
+
+  clear (section, key) {
+    if (typeof section !== 'string') return this
+
+    // clear an entire section
+    if (!key) {
+      delete this._metadata[section]
+      return this
+    }
+
+    // clear a single value from a section
+    if (this._metadata[section]) {
+      delete this._metadata[section][key]
+      return this
+    }
+
+    return this
```

Comment:
Is it part of the spec that these methods return `this` to form a fluent API? I recall also seeing it in previous PRs. Is it necessary/used? I've not seen any test coverage around it as far as I can remember and the type declarations do not expose this fact (at least the new metadata methods in this PR are returning `void`, not `this`). Just wondering if we could get rid and shave a few bytes?

Reply:
Yeah it's not in the spec – I've removed them.

Reply:
Nice, I think there are potentially some more to go back and clean too. I will take a look this week.